### ISSUE TITLE
Sort icon in the TableHeader component is not accessible to screen readers

### ISSUE THREAD

**Issue by ianoxley** [Sage/carbon#1142]
The sort ascending / descending icon is not detected by screen readers, which makes the sorting of table data inaccessible to any such users.

Could we wrap any elements that trigger sorting in a `<a>` tag?
